### DISCUSSION TITLE
Show column for field in related model

### DISCUSSION THREAD

**Discussion by jonocodes** [aminalaee/sqladmin#684] [General]
I have Users and Organizations. I would like in the list view for Users to show the Org name, but that name requires a join across the tables.

I am able to get the column to show, but not able to get the values to display. I have tried using sqlalchemy orm add_columns() and reducing the select to not return a model, but instead return specific fields in list_query().

But I see here
https://github.com/aminalaee/sqladmin/blob/fadf4d6a91143f877c2a7053879202289b163015/sqladmin/models.py#L696

That 'scalars()' is reducing it back to the Users object.

Is there a suggested way for showing joined values like this?

**Comment by aminalaee** [comment]
Yes you can do something like:
‘’’
column_list = [“org.name”]
‘’’

And it will do the join

**Reply by jonocodes** [reply]
Thanks for your reply. That does not work for me. Do you have any examples? Do I need to override list_query() still?
scalars() still reduces the result to a User object, which wipes out the additional column.

**Reply by jonocodes** [reply]
Oh sorry, I was capitalizing 'Org', which of course represents the model not the relationship. User error. Well your suggestion works. Thanks!

**Reply by jonocodes** [reply]
It does look like an n+1 though as it is performing a join for each user, instead of prefetching the relation. It probably the way I defined my models.

**Comment by JakNowy** [comment]
@aminalaee This works great for me, but is it possible to rename the field label back to "name" instead of "org.name"? Or introduce some column aliases in general?

**Reply by aminalaee** [reply]
I think the column_labels should work: https://aminalaee.dev/sqladmin/configurations/#general-options
